MultisigIntegrationTests/SafeTransactionServiceIntegrationTests.swift

```diff
@@ -22,6 +22,34 @@ class SafeTransactionServiceIntegrationTests: XCTestCase {
         semaphore.wait()
     }
 
+    func test_safeInfo_whenSendingNotASafe_returns404Error() {
+        let semaphore = DispatchSemaphore(value: 0)
+        DispatchQueue.global().async {
+            try? XCTAssertThrowsError(try self.service.safeInfo(at: "0xc778417E063141139Fce010982780140Aa0cD5Ab")) {
+                guard case HTTPClient.Error.entityNotFound(_, _, _) = $0 else {
+                    XCTFail("Wrong error type")
+                    return
+                }
+            }
+            semaphore.signal()
+        }
+        semaphore.wait()
+    }
+
+    func test_safeInfo_whenSendingNotNormalizedAddress_returns422Error() {
+        let semaphore = DispatchSemaphore(value: 0)
+        DispatchQueue.global().async {
```

Comment:
I want to prototype re-implementing the networking using Combine, we might get rid of chunks of the code.

Reply:
sounds good